test/integration/dvl_system_water_mass_tracking.cc

```diff
@@ -0,0 +1,189 @@
+    EXPECT_NEAR(linearVelocityEstimate.Y(),
+                expectedLinearVelocityEstimate.Y(), kVelocityTolerance);
+    EXPECT_NEAR(linearVelocityEstimate.Z(),
+                expectedLinearVelocityEstimate.Z(), kVelocityTolerance);
+  }
+
+  // Move the AUV in a straight line
+  // Manipualtor sets linear velocity in body frame
+  fixture.Manipulator().SetLinearVelocity(math::Vector3d::UnitX);
+
+  // Have the AUV describe a circle
+  // todo(anyone) Having a non-zero angular velocity produces inaccurate
+  // velocity estimates. Investigate whether it is a test issue or gz-sensors
+  // dvl implementation issue
+  // fixture.Manipulator().SetAngularVelocity(math::Vector3d::UnitZ);
+
+  // Step simulation for some time for DVL estimates to estabilize
+  fixture.Step(50s);
+
+  ASSERT_TRUE(velocitySubscription.WaitForMessages(50, 10s));
+
+  {
+    const msgs::DVLVelocityTracking message =
+        velocitySubscription.ReadLastMessage();
+    ASSERT_TRUE(message.has_target());
+    const msgs::DVLTrackingTarget & target = message.target();
+    EXPECT_EQ(target.type(), msgs::DVLTrackingTarget::DVL_TARGET_WATER_MASS);
+    for (int i = 0; i < message.beams_size(); ++i)
+    {
+      const msgs::DVLBeamState & beam = message.beams(i);
+      EXPECT_EQ(beam.id(), i + 1);
+      EXPECT_TRUE(beam.locked())
+          << "Beam #" << beam.id() << " not locked";
+    }
+    ASSERT_TRUE(message.has_velocity());
+    const math::Vector3d linearVelocityEstimate =
+        msgs::Convert(message.velocity().mean());
+
+    const auto &linearVelocities =
+        fixture.Observer().LinearVelocities();
+    const auto &angularVelocities =
+        fixture.Observer().AngularVelocities();
+    const auto &poses = fixture.Observer().Poses();
+
+    // Linear velocities w.r.t. to underwater currents
+    // are reported in a sensor affixed, SFM frame.
+
+    // linear velocity output from fixture Observer is in world frame
+    // convert to body frame
+    math::Vector3d linearVelocityBodyFrame =
+        poses.back().Rot().RotateVectorReverse(
+            linearVelocities.back() - waterCurrentVelocity);
+
+    // first get linear velocity of body w.r.t reference frame
+    math::Vector3d linearVelocityRefFrame =
+      bodyToRef.RotateVectorReverse(linearVelocityBodyFrame);
+
+    // get linear velocity at sensor pos w.r.t reference frame
+    // sensor is at a pos offset from body. Compute tangential velocity
+    math::Vector3d angularVelocityBodyFrame =
+        poses.back().Rot().RotateVectorReverse(
+            angularVelocities.back());
+    math::Vector3d angularVelocityRefFrame =
+        bodyToRef.RotateVectorReverse(angularVelocityBodyFrame);
+    math::Vector3d tangentialVelocityRefFrame =
+        angularVelocityRefFrame.Cross(sensorPositionInSFMFrame);
+    // calculate the final linear velocity estimate in reference frame
+    math::Vector3d expectedLinearVelocityEstimate =
+        linearVelocityRefFrame + tangentialVelocityRefFrame;
```

Comment:
@iche033 this seems about right (thanks for the fix BTW, frames for angular velocity contributions were off). If the velocity commands are correct (see the request for clarification above), maybe sensor linear velocities w.r.t. the world frame as observed in the world frame back in the DVL are incorrect? [This code](https://github.com/gazebosim/gz-sensors/blob/0440bec8668da43f6fd84d8c9e5a14b3d679c94c/src/DopplerVelocityLog.cc#L1746-L1751) would be wrong if the reported velocity were not accounting for the transform between body and sensor frame.